docs/architecture/adr-040-storage-and-smt-state-commitments.md

```diff
@@ -110,6 +110,36 @@ We need to be able to process transactions and roll-back state updates if a tran
 
 We identified use-cases, where modules will need to save an object commitment without storing an object itself. Sometimes clients are receiving complex objects, and they have no way to prove a correctness of that object without knowing the storage layout. For those use cases it would be easier to commit to the object without storing it directly.
 
+
+### Low Level Interface
+
+Modules should be able to commit a value fully managed by the module itself. For example, a module can manage its own special database and commit its state by setting a value only to `SC`.
+Similarly, a module can save a value without committing it - this is useful for ORM module or secondary indexes (eg x/staking `UnbondingDelegationKey` and `UnbondingDelegationByValIndexKey`).
+
+Currently, a module can access a store only through `sdk.Context`. We add the following methods to the `sdk.Context`:
+
+```
+type StoreAccess inteface {
+    KVStore(key []byte) KVStore  // reads and writes to combined store (SS & SC)
+    InternalSCStore(key []byte) KVStore  // reads and writes only to the SC
```

Comment:
I'm really thinking about removing a direct access to `SC`. We can always add it later on if we will find a real need. For the moment only use case we are aware about is to commit a private store of a module. So small amount of hashes. This can be done by saving directly to the general store.